src/optiga/optiga.c

```diff
@@ -0,0 +1,1275 @@
+    0x20,
+    // Number of bytes that follow
+    19,
+    // Set LcsO. Refer to macro to see the value or some more notes.
+    0xC0,
+    0x01,
+    FINAL_LCSO_STATE,
+    // Data object type: BSTR.
+    // See table "Data Object Types":
+    // https://github.com/Infineon/optiga-trust-m-overview/blob/98b2b9c178f0391b1ab26b52082899704dab688a/docs/OPTIGA%E2%84%A2%20Trust%20M%20Solution%20Reference%20Manual.md#linkaf9aa284_1397_4161_8761_8c44fbbfa69d
+    0xE8,
+    0x01,
+    0x00,
+    // Allow writes, enforce shielded connection.
+    0xD0,
+    0x03,
+    0x20,
+    0xE1,
+    0x40,
+    // Allow reads, enforce shielded connection.
+    0xD1,
+    0x03,
+    0x20,
+    0xE1,
+    0x40,
+    // Disallow exe
+    0xD3,
+    0x01,
+    0xFF,
+};
+
+static const uint8_t _attestation_metadata[] = {
+    // Metadata tag in the data object
+    0x20,
+    // Number of bytes that follow
+    17,
+    // Set LcsO. Refer to macro to see the value or some more notes.
+    0xC0,
+    0x01,
+    FINAL_LCSO_STATE,
+    // Key usage associated with key container: Sign
+    // See table "Metadata associated with data and key objects":
+    // https://github.com/Infineon/optiga-trust-m-overview/blob/98b2b9c178f0391b1ab26b52082899704dab688a/docs/OPTIGA%E2%84%A2%20Trust%20M%20Solution%20Reference%20Manual.md#link8051b344_ff66_4d6b_bcfd_d21bb87d05d4
+    0xE1,
+    0x01,
+    0x10,
+    // Allow writes - GenKeyPair requires this to be able to write.
+    // However, writes from the host are forbidden.
+    // Table "Common key objects with TAG’s and AC‘s" - 0xE0F1:
+    // https://github.com/Infineon/optiga-trust-m-overview/blob/98b2b9c178f0391b1ab26b52082899704dab688a/docs/OPTIGA%E2%84%A2%20Trust%20M%20Solution%20Reference%20Manual.md#linkf056a0f7_a31f_41c3_b1d9_f270a4fe0378
+    // "The GetDataObject, and SetDataObject commands are not allowed for the data part of the key
+    // object even if the metadata state the access rights differently"
+    0xD0,
+    0x01,
+    0x00,
+    // Disallow reads
+    0xD1,
+    0x01,
+    0xFF,
+    // Execute: enforce shielded connection.
+    // See Table 'Access Condition Identifier and Operators" -> "Conf":
+    // https://github.com/Infineon/optiga-trust-m-overview/blob/98b2b9c178f0391b1ab26b52082899704dab688a/docs/OPTIGA%E2%84%A2%20Trust%20M%20Solution%20Reference%20Manual.md#linkc15dfea4_2cc2_46ae_a53b_1e6ea9487f34
+    0xD3,
+    0x03,
+    0x20,
+    0xE1,
+    0x40,
+};
+
+static const uint8_t _counter_metadata[] = {
+    // Metadata tag in the data object
+    0x20,
+    // Number of bytes that follow
+    16,
+    // Set LcsO. Refer to macro to see the value or some more notes.
+    0xC0,
+    0x01,
+    FINAL_LCSO_STATE,
+    // Change/Write access. This allows updating the counter when LcsO < op.
+    0xD0,
+    0x03,
+    0xE1,
+    0xFC,
+    LCSO_STATE_OPERATIONAL,
+    // Allow reads, enforce shielded connection.
+    0xD1,
+    0x03,
+    0x20,
+    0xE1,
+    0x40,
+    // Allow exe
```

Comment:
should exe also enforce shielded connection?

Reply:
Weirdly, exe needs to be allowed so the counter can be attached to commands on other objects. The counter is then incremented internally. Not sure if this would continue to work with shielded comm, as we don't access it directly.

There are commands to increment it manually, I guess with this config they could be called without a shielded communication, but then again that is not a problem.